bot.py

```diff
@@ -1,10 +1,86 @@
+import asyncio
 import discord
 import secret
-from discord.ext import commands
+import random
+import re
 
+from discord.ext import commands
 from cogs.hello import Greetings
+from MessageProcessor import *
+
+class ChannelStats(MessageProcessor):
+    def __init__(self, channel):
+        self.channel = channel #todo: assert text channel (has name etc)
+        self.replies_queue = []
+
+        #GIF checker processor
+        self.gif_regex = re.compile('\.gif\?*\S*$')
+        self.gif_warning_chance = 0
+    
+    def process(self, message):
+        if message.channel.id != self.channel.id:
+            raise ValueError('Message to be processed does not come from channel {} - src channel ID: {}, src channel type: {}'
+                                .format(self.channel.name, message.channel.id, message.channel.type))
+        super().process(message)
+    
+    async def send_replies(self):
+        for reply in self.replies_queue:
+            await reply
+        self.replies_queue.clear()
+    
+    @processor
+    def check_for_gif(self, message):
```

Comment:
Daloby sie ta metode przepisac by dzialala mniej wiecej tak:

1. sprawdz czy jest gif w jakikolwiek sposob w wiadomosci (ew. policz ilosc)
2. jezeli jest/są to zwieksz warning chance ORAZ daj wiadomosc ostrzegajacą, warning chance /= 2
3. jezeli nie ma to zmniejsz warning chance
4. jezeli warning chance < 0 to warning chance =0

Chyba było by krócej i report embed byłby odpalany tylko raz